org.eclipse.january.examples/src/org/eclipse/january/examples/dataset/AxesMetadataExample.java

```diff
@@ -0,0 +1,338 @@
+                // measurements in the overlapping period (slice() operations may help with this)
+
+                // ok, use a - it's the most frequent
+                useAforTimeMaster = true;
+              }
+              else
+              {
+                // ok, use b - it's the most frequent
+                useAforTimeMaster = false;
+              }
+
+              // ok, decide which way around things are
+              final Dataset indices = useAforTimeMaster ? bIndices : aIndices;
+              final Dataset values = useAforTimeMaster ? db : da;
+              final IDataset masterIndices = useAforTimeMaster ? aIndices
+                  : bIndices;
+              final IDataset masterValues = useAforTimeMaster ? da : db;
+
+              // ok, now do interpolation
+              final Dataset interpolatedValues = Maths.interpolate(indices,
+                  values, masterIndices, null, null);
+              operandA = masterValues;
+              operandB = interpolatedValues;
+
+              // remember the output axes, since we'll put them into the results
+              interpolatedValues.addMetadata(masterValues.getFirstMetadata(
+                  AxesMetadata.class));
+            }
+            else
+            {
+              // an exception was thrown while trying to determine the
+              // indices. See if the parent class can handle it
+              operandA = da;
+              operandB = db;
+            }
+          }
+          else
+          {
+            // no interpolation needed
+            operandA = da;
+            operandB = db;
+          }
+        }
+        else
+        {
+          // we don't know the axes, so forget about syncing
+          operandA = da;
+          operandB = db;
+        }
+      }
+      else
+      {
+        // we can only handle datasets, leave them to it.
+        operandA = a;
+        operandB = b;
+      }
+
+      // perform the basic operation
+      final Dataset res = operation.perform(operandA, operandB, o);
+
+      // ok, inject the axes back into the result, if we can
+      if (operandA instanceof IDataset)
+      {
+        final IDataset ds = (IDataset) operandA;
+        final AxesMetadata targetAxes = ds.getFirstMetadata(AxesMetadata.class);
+        if (targetAxes != null)
+        {
+          res.addMetadata(targetAxes);
+        }
+      }
+
+      return res;
+    }
+
+  }
+
+  @Before
+  public void before()
+  {
+    // prevent missing logger from being reported
+    Utils.suppressSLF4JError();
+  }
+
+  @Test
+  public void testNonSyncedAxesOperationInNewMaths()
+  {
+    // configure Dataset A
+    final Dataset elevationA_m = DatasetFactory.createFromList(Arrays.asList(
+        50l, 100l, 150l));
+    elevationA_m.setName("A elevation (m)");
+    Dataset temperatureA_C = DatasetFactory.createFromList(Arrays.asList(5d,
+        10d, 15d));
+    final AxesMetadata axesMetadataA = new AxesMetadataImpl();
```

Comment:
Ooh, the software engineer in me is uncomfortable here.

I appear to have two choices:
````
AxesMetadata newItem = new AxesMetadataImpl();
newItem.initialize(1);
````
or 
````
AxesMetadata newItem = MetadataFactory.create(AxesMetadataImpl.class, 1);
````
The second one is fewer lines. But, in the first one the compiler verifies that I'm providing the correct arguments to the initialize function.

@PeterC-DLS - Are there other benefits to the second approach that I'm missing?